### PR TITLE
Add cut direction for lawn mower goat g1

### PR DESCRIPTION
**Pull request by Augar**
Extends goat lawn mower settings to adjust the cutting angle

## Summary by CodeRabbit

- **New Features**
  - Introduced cut direction capability, allowing users to get and set the cut direction for their Deebot devices.

- **Bug Fixes**
  - Added comprehensive test cases for new cut direction commands to ensure reliability and correctness.

- **Improvements**
  - Enhanced event handling with the new `CutDirectionEvent` class, improving device interaction and control.

### CODE REVIEW

File: deebot_client/capabilities.py [hunk 1]
```diff
@@ -189,6 +190,7 @@ class CapabilitySettings:
     )
     border_switch: CapabilitySetEnable[BorderSwitchEvent] | None = None
     child_lock: CapabilitySetEnable[ChildLockEvent] | None = None
+    cut_direction: CapabilitySet[CutDirectionEvent, int] | None = None
```
Comment (edenhaus) [inline deebot_client/capabilities.py:193]: I would create a MowerCapabilitySettings class instead, which extends this one. As a vacuum will never have a cut direction

**Review by edenhaus — CHANGES_REQUESTED** [review]

File: deebot_client/capabilities.py (hunk 1 above)
Comment (Augar) [reply deebot_client/capabilities.py]: Thx for the pointer :-)

Comment (Augar) [reply deebot_client/capabilities.py:193]: Done, I've reverted the last commit.

**Review by edenhaus — APPROVED** [review]
Thanks @Augar :+1:

**Pull request merged**